README.md

```diff
@@ -1 +1,113 @@
-# casdoor-nginx-module
\ No newline at end of file
+# casdoor-nginx-module
+
+An NGINX module that provides custom token-based authentication using Casdoor as the backend service.
+
+## Overview
+
+This module implements a **custom token protocol** that communicates with Casdoor using a tab-separated format. It provides two main functions:
+- **Token Generation**: `casdoor_token` directive generates access tokens
+- **Token Validation**: `casdoor_check` directive validates access tokens
+
+## Build NGINX with this module
+
+```bash
+# Configure nginx with the module
+./auto/configure --add-module=/path/to/casdoor-nginx-module
+
+# Build
+make
+
+# (Optional) Install
+make install
+```
+
+Find "ngx_http_casdoor_module" in objs/ngx_modules.c to verify the module is included.
+
+## Setup Casdoor
+
+### Build and Run Casdoor
+
+```bash
+git clone https://github.com/casdoor/casdoor
+cd casdoor
+# Build frontend
+cd web && yarn install && yarn build && cd ..
+# Run casdoor server
+go run main.go
+```
+
+Casdoor will run on `http://localhost:8000` by default.
+
+For more details, see the official documentation: [Casdoor](https://casdoor.org/docs/deployment/nginx/)
+
+## Module Directives
+
+### Configuration Directives
+- `casdoor_appid "<application_id>"`: Casdoor application ID
+- `casdoor_secret "<application_secret>"`: Casdoor application secret  
```

Comment:
You need to know Casdoor SDK's "classic" args: https://github.com/casdoor/casdoor-go-sdk?tab=readme-ov-file#step-2-init

I don't know how you can work with the missing args.

<img width="2000" height="1258" alt="image" src="https://github.com/user-attachments/assets/a43fe6d5-e9ef-4947-b766-d74eb70ab22f" />